### PR TITLE
S3

### PR DESCRIPTION
**Pull request by heron2014**
Related to #67

### CODE REVIEW

File: src/lib/auth/update-user-photo.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import query from '../../db/query';
+import { updateUserPhoto as queryText } from '../../db/sql-queries.json';
+
+/**
+ * updateUserPhoto updates user photo url in the database
+ * @param {object} client - database client
+ * @param {string} user_id - user_id
+ * @param {string} filename - new photo_url
+ * @returns {Promise.<void, Error>}
```
Comment (minaorangina) [inline src/lib/auth/update-user-photo.js]: I've noticed we've been copying/pasting this line wrong in a lot of our code.
In this case,`{Promise.<void, error>}` should be `{Promise.<string, error>}` if this function returns a string (does it?).

Keep an eye out of other times we've done this, I'm trying to correct them as and when I see it

File: src/lib/handlers.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+import formidable from 'formidable';
+import gm from 'gm';
+import s3 from './s3-client';
+
 import { UPDATE_FEED } from '../../socket-router';
 import saveEvent from './events/save-event';
 import getEvent from './events/get-event';
 import getUserById from './auth/get-user-by-id';
 import updateUser from './auth/update-user';
+import updateUserPhoto from './auth/update-user-photo'; //eslint-disable-line no-unused-vars
```
Comment (minaorangina) [inline src/lib/handlers.js]: don't need to disable this line anymore

File: src/lib/handlers.js [hunk 3]
```diff
@@ -17,6 +24,8 @@ import buildFeedItem from './events/build-feed-item';
 import normaliseEventKeys from './normalise-event-keys';
 import client from '../db/client';
 import shortid from 'shortid';
+import generateFileName from './generate-file-name';
+import extractFileExtension from './extract-file-extension';
 
 export function postEventHandler (req, res, next) { // eslint-disable-line no-unused-vars
```
Comment (minaorangina) [inline src/lib/handlers.js]: think this line doesn't need to be ignored either

File: src/lib/s3-client.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import aws from 'aws-sdk'; //eslint-disable-line
```
Comment (minaorangina) [inline src/lib/s3-client.js]: doesn't need to be ignored

File: src/lib/handlers.js (hunk 3 above)
Comment (heron2014) [inline src/lib/handlers.js]: @minaorangina this ignore statement is for next param - we are not using next in this function.

File: src/lib/auth/update-user-photo.js (hunk 1 above)
Comment (heron2014) [inline src/lib/auth/update-user-photo.js]: @minaorangina this function returns an object { photo_url: 'https//...'}

Comment (heron2014) [inline src/lib/auth/update-user-photo.js]: I guess that should be {Promise <object, error>}

**Pull request merged**